### PR TITLE
Laboratory work #1, Sofya Borodina - 22FPL1

### PR DESCRIPTION
**Pull request by Sofya0borodina**

### CODE REVIEW

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -5,20 +5,29 @@
 
 
 def tokenize(text: str) -> list[str] | None:
-    """
-    Splits a text into tokens, converts the tokens into lowercase,
-    removes punctuation, digits and other symbols
-    :param text: a text
-    :return: a list of lower-cased tokens without punctuation
-    """
```
Comment (marina-kaz) [inline lab_1_classify_by_unigrams/main.py:13]: Please do not delete docstrings!

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -5,20 +5,29 @@
...
+    text = text.lower()
+    tokens = []
+    for symbol in text:
+        if symbol.isalpha():
+            tokens.append(symbol)
+    return tokens
 
 
-def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
+def calculate_frequencies(tokens: tokenizelist[str] | None) -> dict[str, float] | None:
```
Comment (marina-kaz) [inline lab_1_classify_by_unigrams/main.py:16]: what is `tokenizelist`?

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -5,20 +5,29 @@
...
     Calculates frequencies of given tokens
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    frequencies = {}
+    for symbol in tokens:
+        if symbol not in frequencies:
+            frequencies[symbol] = 1
+        else:
+            frequencies[symbol] = frequencies[symbol] + 1
```
Comment (marina-kaz) [inline lab_1_classify_by_unigrams/main.py:27]: Consider using `+=` expression

**Review by marina-kaz — CHANGES_REQUESTED** [review]
Good start! 

Please also don't forget to include checks of input arguments. What if they are incorrect?

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
Comment (Sofya0borodina) [reply lab_1_classify_by_unigrams/main.py:13]: Fixed

File: lab_1_classify_by_unigrams/main.py (hunk 2 above)
Comment (Sofya0borodina) [reply lab_1_classify_by_unigrams/main.py]: Fixed

File: lab_1_classify_by_unigrams/main.py (hunk 3 above)
Comment (Sofya0borodina) [reply lab_1_classify_by_unigrams/main.py]: Fixed

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -19,6 +28,15 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    frequencies = {}
```
Comment (marina-kaz) [inline lab_1_classify_by_unigrams/main.py:31]: Don't forget to add checks for incorrect types here and in other functions.

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -37,6 +57,13 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
     :param actual: a list of actual values
     :return: the score
     """
+    number = 0
+    total = 0
+    for i, value in enumerate(actual):
+        total += (value - predicted[i])**2
+        number += 1
+    score = round(total/number, 3)
```
Comment (marina-kaz) [inline lab_1_classify_by_unigrams/main.py:65]: Is this the right place for rounding?

File: lab_1_classify_by_unigrams/start.py [hunk 6]
```diff
@@ -9,12 +10,16 @@ def main() -> None:
...
+        en_profile = create_language_profile("en", en_text)
     with open("assets/texts/de.txt", "r", encoding="utf-8") as file_to_read_de:
         de_text = file_to_read_de.read()
+        de_profile = create_language_profile("de", de_text)
     with open("assets/texts/unknown.txt", "r", encoding="utf-8") as file_to_read_unk:
         unknown_text = file_to_read_unk.read()
+        unk_profile = create_language_profile("unk", unk_text)
     result = None
     assert result, "Detection result is None"
+    print(detect_language(unk_profile, en_profile, de_profile))
```
Comment (marina-kaz) [inline lab_1_classify_by_unigrams/start.py:22]: Does this line ever get printed?

**Review by marina-kaz — CHANGES_REQUESTED** [review]
No changes since my last review

**Review by marina-kaz — COMMENTED** [review]
Nothing to review today

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -49,6 +76,20 @@ def compare_profiles(
...
+    values_unk = unknown_profile["freq"]
+    values_comp = profile_to_compare["freq"]
+    for k in values_unk:
+        if k not in values_comp:
+            values_comp[k] = 0
+    for k in values_comp:
+        if k not in values_unk:
+            values_unk[k] = 0
+    sorted_values_unk = dict(sorted(values_unk.items()))
+    sorted_values_comp = dict(sorted(values_comp.items()))
```
[the remark this answers is not in the document]
Comment (Sofya0borodina) [reply lab_1_classify_by_unigrams/main.py:88]: Without sorting this dictionaries the program gives wrong results.

File: lab_1_classify_by_unigrams/main.py (hunk 5 above)
Comment (Sofya0borodina) [reply lab_1_classify_by_unigrams/main.py:65]: Fixed.

File: lab_1_classify_by_unigrams/main.py [hunk 8]
```diff
@@ -19,6 +28,15 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    frequencies = {}
+    for symbol in tokens:
+        if symbol not in frequencies:
+            frequencies[symbol] = 1
+        else:
```
[the remark this answers is not in the document]
Comment (Sofya0borodina) [reply lab_1_classify_by_unigrams/main.py:35]: Done.

File: lab_1_classify_by_unigrams/main.py (hunk 4 above)
Comment (Sofya0borodina) [reply lab_1_classify_by_unigrams/main.py:31]: Done.

File: lab_1_classify_by_unigrams/main.py [hunk 9]
```diff
@@ -11,6 +11,15 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    try:
+        text = text.lower()
+    except TypeError:
```
[the remark this answers is not in the document]
Comment (Sofya0borodina) [reply lab_1_classify_by_unigrams/main.py:16]: Fixed.

File: lab_1_classify_by_unigrams/start.py (hunk 6 above)
Comment (Sofya0borodina) [reply lab_1_classify_by_unigrams/start.py]: Fixed

File: lab_1_classify_by_unigrams/main.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (marina-kaz) [reply lab_1_classify_by_unigrams/main.py:39]: Task: try again

**Pull request closed without merging**